src/components/ToDoContainer/index.js

```diff
@@ -1,11 +1,77 @@
-import React, { Component } from "react";
+import React, {
+  Component
+} from "react";
+
+import NewTodoForm from './NewTodoForm';
+import TodoList from './TodoList';
 
 class ToDoContainer extends Component {
+  constructor() {
+    super();
+    this.state = {
+      newTodo: '',
+      todos: []
+    }
+
+    this.formSubmitted = this.formSubmitted.bind(this);
+    this.newTodoChanged = this.newTodoChanged.bind(this);
+    this.removeTodo = this.removeTodo.bind(this);
+    this.searchTodos = this.searchTodos.bind(this);
+  }
+
+  // Create new todo
+  newTodoChanged(event) {
+    this.setState({
+      newTodo: event.target.value
+    })
+
+  }
+
+  // Form submission event handler
+  formSubmitted(event) {
+    event.preventDefault();
+    // Add new todo to array
+    this.setState({
+      newTodo: '',
+      todos: [...this.state.todos, {
+        title: this.state.newTodo,
+        done: false
+      }]
+    })
+  }
+
+  // Remove clicked todolist
+  removeTodo(index) {
```

Comment:
Consider an alternate method of removing todos from the list than index / position in the list and splicing on that - splice is a bit messy here since it causes side-effects on the `todos` variable. Instead, you could store the index of each each todo in `this.state.todos` as an `id` property in your todo object. Then, when you call `removeTodo` with a given index, instead of splicing, you can use the cleaner `.filter()` method to exclude the todo with that index as its `id` and then set the state to that updated array.

Reply:
Tried this approach but it causes this bug where once I update state with the filtered array the indexes are modified due to the removal of one element such that when I try to delete the second element it doesn't work. Fixed it by changing the ID everytime I do the filtering but I don't think it's the most elegant solution.